endpoints/github.ts

```diff
@@ -111,17 +113,29 @@ const statesColor = {
   unknown: 'grey'
 }
 
-function combineState (states: Array<any>) {
+function combined (states: Array<any>, stateKey?: string) {
+  const k = stateKey || 'state'
   if (states.length === 0) return 'unknown'
-  if (states.find(x => x.state === 'error')) return 'error'
-  if (states.find(x => x.state === 'failure')) return 'failure'
-  if (states.find(x => x.state === 'pending')) return 'pending'
-  if (states.every(x => x.state === 'success')) return 'success'
+  if (states.find(x => x[k] === 'error')) return 'error'
+  if (states.find(x => x[k] === 'failure')) return 'failure'
+  if (states.find(x => x[k] === 'pending')) return 'pending'
+  if (states.every(x => x[k] === 'success')) return 'success'
```

Comment:
I mean, we can give a default value to `k` in the arguments:

```js
function combined (states: Array<any>, stateKey = 'state') {
  if (states.length === 0) return 'unknown'
  if (states.find(x => x[stateKey] === 'error')) return 'error'
  if (states.find(x => x[stateKey] === 'failure')) return 'failure'
  if (states.find(x => x[stateKey] === 'pending')) return 'pending'
  if (states.every(x => x[stateKey] === 'success')) return 'success'
```

just like you did in the `checks()` function:

```js
async function checks ({ owner, repo, ref = 'master'}: Args) {
```

Reply:
Yea we can.

Reply:
If it has a default value, then it is optional 😉

![image](https://user-images.githubusercontent.com/215282/67454350-79a82e80-f65c-11e9-878f-797e0dae3719.png)

https://www.typescriptlang.org/play/index.html?ssl=3&ssc=1&pln=3&pc=13#code/GYVwdgxgLglg9mABBOBbARjMBTAJgCgGcoBDKbQgLkQEEAnOkgTwB4SwmA+AGkWLOwBpbE0QBeRAHJ+5SQEpEAb0QBfALAAoTSgxY8+ANoBdOUA